internal/plugin/resolver.go

```diff
@@ -81,30 +81,70 @@ func (resolver *Resolver) Resolve(credentials []*config_v2.Credential) (map[stri
 	errorStrings := make([]string, 0, len(credentials))
 
 	var err error
+
+	// Group credentials by provider
+	var credentialsByProvider = make(map[string][]*config_v2.Credential)
 	for _, credential := range credentials {
-		var provider plugin_v1.Provider
-		var value []byte
+		credentialsByProvider[credential.From] = append(
+			credentialsByProvider[credential.From],
+			credential,
+		)
+	}
 
-		if provider, err = resolver.Provider(credential.From); err != nil {
-			resolver.LogFatalf("ERROR: Provider '%s' could not be used! %v", credential.From, err)
+	// Resolve credentials by provider
+	for providerID, credentialsForProvider := range credentialsByProvider {
+		provider, err := resolver.Provider(providerID)
+		if err != nil {
+			resolver.LogFatalf("ERROR: Provider '%s' could not be used! %v", providerID, err)
 		}
 
-		// This provider cannot resolve the named credential
-		if value, err = provider.GetValue(credential.Get); err != nil {
-			errInfo := fmt.Sprintf("ERROR: Resolving credential '%s' from provider '%s' failed: %v",
-				credential.Get,
-				credential.From,
-				err)
+		// Create secretIds slice and credentialBySecretId map
+		var secretIds = make([]string, len(credentialsForProvider))
+		for idx, cred := range credentialsForProvider {
+			secretIds[idx] = cred.Get
+		}
+
+		// Resolves all credentials for current provider
+		providerResponses, err := provider.GetValues(secretIds...)
+		if err != nil {
+			errInfo := fmt.Sprintf(
+				"ERROR: Resolving credentials from provider '%s' failed: %v",
+				provider.GetName(),
+				err,
+			)
 			log.Println(errInfo)
 
 			errorStrings = append(errorStrings, errInfo)
 			continue
 		}
 
-		result[credential.Name] = value
+		// Collect errors from provider responses
+		var hasErrors bool
+		for _, providerResponse := range providerResponses {
+			if providerResponse.Error != nil {
+				hasErrors = true
+				errorStrings = append(errorStrings, providerResponse.Error.Error())
+				continue
+			}
+		}
+		if hasErrors {
+			continue
+		}
+
+		// Set provider responses on result map before returning
+		for _, credential := range credentialsForProvider {
+			credentialName := credential.Name
+			secretValue := providerResponses[credential.Get].Value
+
+			result[credentialName] = secretValue
 
-		if resolver.EventNotifier != nil {
-			resolver.EventNotifier.ResolveCredential(provider, credential.Name, value)
+			if resolver.EventNotifier != nil {
+				resolver.EventNotifier.ResolveCredential(
```

Comment:
This seems like a very late time to notify that the credentials changed. I think we need to do this before sending the credential to a provider - WDYT?

Reply:
You need the credential value to notify so it needs to be after getting the credential from the provider.

Reply:
You are only notifying the management module and we don't want to pass the cred value so I think it can be done earlier

Reply:
@sgnn7 You do pass the value `resolver.EventNotifier.ResolveCredential(provider, credential.Name, value)`

Reply:
Hmm... this is an odd one - I think the API definition is not quite right maybe. There is no reason that I can think of for why a management module would need to know about the value of the secret and we're not using it as a callback for the connector to get the value. For this PR, we can ignore it but it's something that probably should be figured out.